irksome/dirk_stepper.py

```diff
@@ -180,23 +128,29 @@ def __init__(self, F, butcher_tableau, t, dt, u0, bcs=None,
             self.problem, appctx=appctx, solver_parameters=solver_parameters,
             nullspace=nullspace)
 
-        self.kgac = k, g, a, c
+        self.kgac = k, g, a, c, a_vals, d_val
+
+    def update_bc_constants(self, i, a_vals, d_val, c):
+        AAb = self.AAb
+        CCone = self.CCone
```

Comment:
This should do if `self.butcher_tableau` is a property of the class
```suggestion
        shift = 1 if self.butcher_tableau.is_explicit else 0
        AAb = self.AAb[shift:]
        CCone = self.CCone[shift:]
```

Reply:
If we do this (and I'd be fine to do it, I'd say that we do this in the `__init__` method, rather than here, since we don't need to do this at every stage of every timestep

Reply:
Now done in the `__init__` method, with comments to document the logic